core/src/cluster_info.rs

```diff
@@ -108,6 +108,53 @@ pub struct DataBudget {
                   // used to detect when to up the bytes budget again
 }
 
+#[derive(Default)]
+struct Counter(AtomicU64);
+
+impl Counter {
+    fn add_m(&self, x: &mut Measure) {
+        x.stop();
+        self.0.fetch_add(x.as_us(), Ordering::Relaxed);
+    }
+    fn add_r(&self, x: u64) {
```

Comment:
`_r` means what here? Callable from read-only? Relaxed ordering? Can you give these better names or add code comments?